src/weather_model_graphs/create/mesh/mesh.py

```diff
@@ -94,36 +94,34 @@ def create_multirange_2d_mesh_graphs(
     max_num_levels : int
         Number of edge-distance levels in mesh graph
     xy : np.ndarray
-        Grid point coordinates, shaped [2, M, N]
-    refinement_factor : int
-        Degree of refinement between successive mesh graphs, the number of nodes
-        grows by approximately refinement_factor**2 between successive
-        mesh graphs.
+        Grid point coordinates, shaped [N_grid_points, 2]
+    mesh_node_distance: float
+        Distance (in x- and y-direction) between created mesh nodes,
+        in coordinate system of xy
+    level_refinement_factor: float
+        Refinement factor between grid points and bottom level of mesh hierarchy
 
     Returns
     -------
     G_all_levels : list of networkx.Graph
         List of networkx graphs for each level representing the connectivity
         of the mesh within each level
     """
-    # Compute the size (grid nodes) along x and y direction of area
-    # to cover with graph
-    coord_extent = np.array((xy.shape[2], xy.shape[1]))
+    # Compute the size along x and y direction of area to cover with graph
+    # This is measured in the Cartesian coordiantes of xy
+    coord_extent = np.ptp(xy, axis=0)
+    extent_nodes_bottom_mesh = (coord_extent / mesh_node_distance).astype(int)
 
     # Find the number of mesh levels possible in x- and y-direction,
     # and the number of leaf nodes that would correspond to
-    # max_coord/(grid_refinement_factor*level_refinement_factor^mesh_levels) = 1
-    max_mesh_levels_float = (
-        np.log(coord_extent) - np.log(grid_refinement_factor)
-    ) / np.log(level_refinement_factor)
-
-    # Need to add a small epsilon before flooring to int, due to numerical
-    # issues with the computation above
-    eps = 1e-8
-    max_mesh_levels = (max_mesh_levels_float + eps).astype(int)  # (2,)
-    nleaf = grid_refinement_factor * (
-        level_refinement_factor**max_mesh_levels
-    )  # leaves at the bottom in each direction, if using max_mesh_levels
+    # max_mesh_coord/(level_refinement_factor^mesh_levels) = 1
+    max_mesh_levels_float = np.log(extent_nodes_bottom_mesh) / np.log(
```

Comment:
Hmm, I think it is correct. With $m$ = max_mesh _coord, $r$ = level_refinement_factor and $l$ = mesh_levels (all positive):

$$
\frac{m}{r^l} = 1 \Leftrightarrow \log(m) - l \log(r) = 0 \Leftrightarrow l = \frac{\log(m)}{\log(r)}
$$

What is a bit confusing though is that the variable names are not the same as in the comment, so that could be improved.

Reply:
ok, yes, that was why I couldn't get it to line up. But the maths checks out! Thank you for writing that